<launch>

    <!-- search strategy waypoints (Convenience for template generation)-->
    <arg name="point_num" value="3" />

    <arg name="point0_x" value="3.5" />
    <arg name="point0_y" value="0.0" />
    <arg name="point0_z" value="0.7" />

    <arg name="point1_x" value="3.5" />
    <arg name="point1_y" value="1.5" />
    <arg name="point1_z" value="0.7" />

    <arg name="point2_x" value="0.0" />
    <arg name="point2_y" value="1.5" />
    <arg name="point2_z" value="0.7" />

    <arg name="point3_x" value="0.0" />
    <arg name="point3_y" value="0.0" />
    <arg name="point3_z" value="0.7" />

    <arg name="point4_x" value="4.2" />
    <arg name="point4_y" value="-2.2" />
    <arg name="point4_z" value="1.0" />

    <node pkg="search_plan" name="search_plan_node" type="search_node" output="screen">
        <param name="odom_topic" value="/vins_fusion/imu_propagate" type="string"/>
        <param name="exec_frequency" value="100" type="double"/>
        <param name="arrive_threshold" value="0.3" type="double"/>
        <param name="publish_target_threshold" value="0.2" type="double"/>
        <param name="target_msg_timeout" value="5.0" type="double"/>
        <param name="target_converge_th" value="0.10" type="double"/>
        <param name="slow_down_time_duration" value="5.0" type="double"/>
        <param name="slow_down_height" value="1.0" type="double"/>
        <param name="my_target_hover_height" value="1.0" type="double"/>

        <param name="search_startpoint_x" value="1.0" type="double"/>
        <param name="search_startpoint_y" value="0.0" type="double"/>
        <param name="search_startpoint_z" value="0.7" type="double"/>

        <param name="fsm/waypoint_num" value="$(arg point_num)" type="int"/>
        <param name="fsm/waypoint0_x" value="$(arg point0_x)" type="double"/>
        <param name="fsm/waypoint0_y" value="$(arg point0_y)" type="double"/>
        <param name="fsm/waypoint0_z" value="$(arg point0_z)" type="double"/>
        <param name="fsm/waypoint1_x" value="$(arg point1_x)" type="double"/>
        <param name="fsm/waypoint1_y" value="$(arg point1_y)" type="double"/>
        <param name="fsm/waypoint1_z" value="$(arg point1_z)" type="double"/>
        <param name="fsm/waypoint2_x" value="$(arg point2_x)" type="double"/>
        <param name="fsm/waypoint2_y" value="$(arg point2_y)" type="double"/>
        <param name="fsm/waypoint2_z" value="$(arg point2_z)" type="double"/>
        <param name="fsm/waypoint3_x" value="$(arg point3_x)" type="double"/>
        <param name="fsm/waypoint3_y" value="$(arg point3_y)" type="double"/>
        <param name="fsm/waypoint3_z" value="$(arg point3_z)" type="double"/>
        <param name="fsm/waypoint4_x" value="$(arg point4_x)" type="double"/>
        <param name="fsm/waypoint4_y" value="$(arg point4_y)" type="double"/>
        <param name="fsm/waypoint4_z" value="$(arg point4_z)" type="double"/>

    </node>
</launch>